app/controllers/users_controller.rb

```diff
@@ -48,4 +56,14 @@ def authorise
   def user_params
     params.require(:user).permit(:email, :notify_new)
   end
+
+  def accept_cla
+    @user.update!(cla_accepted: true)
+
+    # TODO this takes a while
+    # Can the proposal states be updated in the background?
+    @user.proposed.each do |pr|
```

Comment:
@Floppy any idea how to make this happen in the background or is it not worth worrying about? (There's a noticeable delay, especially if you have lots of PRs open but that's unlikely in normal circumstances I guess.)

Reply:
Yeah, there's a background job runner that will do it. If you do `UpdateProposalJob.perform_later pr.number` I think that will work.

Reply:
Nice, that was a lot easier than I was expecting! Thanks.

Reply:
Helps that I had basically the exact same problem a while back :)